### PR TITLE
Optional blocking in Keithley 2450

### PR DESCRIPTION
**Pull request by aaronsharpe**
Added a parameter to control whether or not ramp commands sent to Keithley 2450s block

### CODE REVIEW

File: src/qcodes/instrument_drivers/Keithley/Keithley_2450.py [hunk 1]
```diff
@@ -472,6 +473,24 @@ def __init__(self, parent: "Keithley2450", name: str, proper_function: str) -> N
             "when making a measurement.",
         )
 
+        self.add_parameter(
+            "block_during_ramp",
+            get_cmd=self._block,
+            set_cmd=self._set_block,
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/Keithley_2450.py:479]: If you replace the values for get_cmd and set_cmd with None here the parameter is a so called manual parameter that holds the state internally and you can avoid the _set_block attribute and simply use the  value directly

File: src/qcodes/instrument_drivers/Keithley/Keithley_2450.py [hunk 2]
```diff
@@ -472,6 +473,24 @@ def __init__(self, parent: "Keithley2450", name: str, proper_function: str) -> N
...
+            "execution of subsequent code. This parameter allows _proper_function"
+            "to either block or not.",
+        )
+
+    def _set_block(self, value: bool) -> None:
+        self._block = value
+
+    def _set_proper_function(self, value: float) -> None:
+        self.write(f"SOUR:{self._proper_function} {value}")
+        if self._block:
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/Keithley_2450.py:491]:
```suggestion
        if self.block_during_ramp():
```

**Comment by jenshnielsen** [comment]
>  Create a file in the docs\changes\newsfragments folder with a short description of the change.

Could you create a file called improved.5547 as noted here with a description of the fix?

**Comment by aaronsharpe** [comment]
Added the requested changes. Sorry for missing the newsfragments doc!

File: src/qcodes/instrument_drivers/Keithley/Keithley_2450.py [hunk 3]
```diff
@@ -472,6 +472,22 @@ def __init__(self, parent: "Keithley2450", name: str, proper_function: str) -> N
...
+            vals=Bool(),
+            docstring="Setting the source output level alone cannot block the "
+            "execution of subsequent code. This parameter allows _proper_function"
+            "to either block or not.",
+        )
+
+    def _set_proper_function(self, value: float) -> None:
+        self.write(f"SOUR:{self._proper_function} {value}")
+        if self.block_during_ramp():
+            self.ask('*OPC?')
```
Comment (jenshnielsen) [inline src/qcodes/instrument_drivers/Keithley/Keithley_2450.py:489]:
```suggestion
            self.ask("*OPC?")
```

**Review by jenshnielsen — APPROVED** [review]

**Pull request merged**